dblookupext/esdtSupply/logsGetter.go

```diff
@@ -0,0 +1,95 @@
+package esdtSupply
+
+import (
+	"encoding/hex"
+
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type logsGetter struct {
+	logsStorer  storage.Storer
+	marshalizer marshal.Marshalizer
+}
+
+func newLogsGetter(
+	marshalizer marshal.Marshalizer,
+	logsStorer storage.Storer,
+) *logsGetter {
+	return &logsGetter{
+		logsStorer:  logsStorer,
+		marshalizer: marshalizer,
+	}
+}
+
+func (lg *logsGetter) getLogsBasedOnBody(blockBody data.BodyHandler) (map[string]data.LogHandler, error) {
+	body, ok := blockBody.(*block.Body)
+	if !ok {
+		return nil, errCannotCastToBlockBody
+	}
+
+	logsDB := make(map[string]data.LogHandler)
+	for _, mb := range body.MiniBlocks {
+		shouldIgnore := mb.Type != block.TxBlock && mb.Type != block.SmartContractResultBlock
+		if shouldIgnore {
+			continue
+		}
+
+		dbLogsMb := lg.getLogsBasedOnMB(mb)
```

Comment:
why do we need this ? in case of indexer - the node already prepares the complete logs on commitBlock. can't we use the input as full logs directly ?

Reply:
This is needed in case of rollback. When method RevertBlock is called I have `only header and body` based on the transactions hashes from miniblocks I can read the logs from storage and revert the changes for the current header.